lib/internal/modules/esm/resolve.js

```diff
@@ -357,11 +388,12 @@ function resolveExportsTarget(
     }
     for (let i = 0; i < keys.length; i++) {
       const key = keys[i];
-      if (key === 'node' || key === 'import' || key === 'default') {
+      if (key === 'default' || conditions.has(key)) {
```

Comment:
```suggestion
      if (key === 'default' || primordials.SetPrototypeHas(conditions, key)) {
```

Reply:
this isn't needed if `conditions` is a SafeSet, i think - unless `conditions` is coming from user loaders, in which case, this is definitely needed